src/recipes/recipes.service.ts

```diff
@@ -40,9 +44,20 @@ export class RecipesService {
   async addRecipe(
     authorId: string,
     addRecipeDto: AddRecipeDto,
+    file?: Express.Multer.File,
   ): Promise<GetRecipeResponse> {
+    const fileUploadResult =
+      file &&
+      (await this.filesService.uploadSingleFile({
+        file,
+        uploadFileDto: { type: 'recipe', isPublic: false },
+      }));
```

Comment:
it would be clearer what it does if it would be extracted to separated method ;)